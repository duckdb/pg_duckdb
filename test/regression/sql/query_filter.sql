CREATE TABLE query_filter_int(a INT);
INSERT INTO query_filter_int SELECT g FROM generate_series(1,100) g;
SELECT COUNT(*) FROM query_filter_int WHERE a  <= 50;
DROP TABLE query_filter_int;

CREATE TABLE query_filter_float(a FLOAT8);
INSERT INTO query_filter_float VALUES (0.9), (1.0), (1.1);
SELECT COUNT(*) FROM query_filter_float WHERE a < 1.0;
SELECT COUNT(*) FROM query_filter_float WHERE a <= 1.0;
SELECT COUNT(*) FROM query_filter_float WHERE a < 1.1;

DROP TABLE query_filter_float;
