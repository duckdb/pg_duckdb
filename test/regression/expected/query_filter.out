CREATE TABLE query_filter_int(a INT);
INSERT INTO query_filter_int SELECT g FROM generate_series(1,100) g;
SELECT COUNT(*) FROM query_filter_int WHERE a  <= 50;
 count 
-------
    50
(1 row)

DROP TABLE query_filter_int;
CREATE TABLE query_filter_float(a FLOAT8);
INSERT INTO query_filter_float VALUES (0.9), (1.0), (1.1);
SELECT COUNT(*) FROM query_filter_float WHERE a < 1.0;
 count 
-------
     1
(1 row)

SELECT COUNT(*) FROM query_filter_float WHERE a <= 1.0;
 count 
-------
     2
(1 row)

SELECT COUNT(*) FROM query_filter_float WHERE a < 1.1;
 count 
-------
     2
(1 row)

DROP TABLE query_filter_float;
CREATE TABLE query_filter_varchar(a VARCHAR);
INSERT INTO query_filter_varchar VALUES ('t1'), ('t2'), ('t1');
SELECT COUNT(*)FROM query_filter_varchar WHERE a = 't1';
 count 
-------
     2
(1 row)

SELECT COUNT(a) FROM query_filter_varchar WHERE a = 't1';
 count 
-------
     2
(1 row)

SELECT a, COUNT(*) FROM query_filter_varchar WHERE a = 't1' GROUP BY a;
 a  | count 
----+-------
 t1 |     2
(1 row)

DROP TABLE query_filter_varchar;
CREATE TABLE query_filter_output_column(a INT, b VARCHAR, c FLOAT8);
INSERT INTO query_filter_output_column VALUES (1, 't1', 1.0), (2, 't1', 2.0), (2, 't2', 1.0);
-- Projection ids list will be used (column `a`is not needed after scan)
SELECT b FROM query_filter_output_column WHERE a = 2;
 b  
----
 t1
 t2
(2 rows)

-- Column ids list used because both of fetched column are used after scan
SELECT a, b FROM query_filter_output_column WHERE b = 't1';
 a | b  
---+----
 1 | t1
 2 | t1
(2 rows)

-- Column ids list used because both of fetched column are used after scan.
-- Swapped order of table columns.
SELECT b, a FROM query_filter_output_column WHERE b = 't1';
 b  | a 
----+---
 t1 | 1
 t1 | 2
(2 rows)

-- Projection ids list will be used (column `b`is not needed after scan)
SELECT a, c FROM query_filter_output_column WHERE b = 't1';
 a | c 
---+---
 1 | 1
 2 | 2
(2 rows)

-- All columns in tuple unordered
SELECT c, a, b FROM query_filter_output_column WHERE a = 2;
 c | a | b  
---+---+----
 2 | 2 | t1
 1 | 2 | t2
(2 rows)

DROP TABLE query_filter_output_column;
